includes/compatibility/class-wcpdf-compatibility-third-party-plugins.php

```diff
@@ -143,6 +146,40 @@ public function add_product_bundles_classes ( $classes, $document_type, $order,
 		return $classes;
 	}
 
+
+	/**
+	 * WPC Product Bundles
+	 * @param string $classes       CSS classes for item row (tr) 
+	 * @param string $document_type PDF Document type
+	 * @param object $order         WC_Order order
+	 * @param int    $item_id       WooCommerce Item ID
+	 */
+	public function add_wpc_product_bundles_classes ( $classes, $document_type, $order, $item_id = '' ) {
+		if ( !class_exists('WPCleverWoosb') ) {
+			return $classes;
+		}
+
+		$item_id = !empty($item_id) ? $item_id : $this->get_item_id_from_classes( $classes );
+		if ( empty($item_id) ) {
```

Comment:
WPCS

Reply:
i changed this to 

`$item_id = !empty( $item_id ) ? $item_id : $this->get_item_id_from_classes( $classes );

		if ( empty( $item_id ) ) {`
		
Was that the issue?

Reply:
yes, although you also added a linebreak that doesn't need to be there and you forgot the line before that too (`!empty($item_id)`)

Reply:
how're those edits?

Reply:
you actually made it worse with your last edit 😆 

You changed this`(which you had corrected in your previous commit, so was all good!)
```php
if ( empty( $item_id ) ) {
```
back to:
```php
if ( empty($item_id) ) {
```

and left the wrong one untouched:
```php
$item_id = !empty($item_id)
```

There's another one in the conversation further down.

Reply:
think i fixed it this time

Reply:
not yet, both L128 and L162 are missing a space (see my other open comment too)!
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/c2686826e13b476e815d14c4c8993bdbfc738746/includes/compatibility/class-wcpdf-compatibility-third-party-plugins.php#L128
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/c2686826e13b476e815d14c4c8993bdbfc738746/includes/compatibility/class-wcpdf-compatibility-third-party-plugins.php#L162

Reply:
i've more a few corrections for extra spaces and the spelling for 'Chained' but i'm not seeing/understanding where the missing spaces are in the above comment :/

I've been installing the WPCS for PHPCodeSniffer to help me with the syntax overall but it's giving several errors.

Reply:
`!empty()`  should be `! empty()`